File: 비타알고/DonggyunKim00/인싸가 되고 싶은 민수.js
const fs = require('fs');

const filePath = process.platform === 'linux' ? '/dev/stdin' : 'input.txt';
const input = fs
  .readFileSync(filePath, 'utf8')
  .toString()
  .trim()
  .split(' ')
  .map(Number);

const divisor = (num) => {
  const yaksu = new Set();
  for (let i = 1; i <= Math.sqrt(num); i++) {
    if (num % i === 0) {
      const multiplied = num / i;
      i !== 1 && yaksu.add(i);
      yaksu.add(multiplied);
    }
  }

  return yaksu;
};

const solution = (input) => {
  const [A, B] = input;
  let obj = {};

  for (let i = A; i <= B; i++) {
    const yaksu = divisor(i);

    yaksu.forEach((value) => {
      if (!obj[value]) {
        obj[value] = 0;
      }
      obj[value] += 1;
    });
  }

  return obj;
};

console.log(Object.keys(solution(input))[0]);

File: 비타알고/JunYoungKr/인싸가되고싶은민수.py
a, b = map(int, input().split())

# a와 b가 다르면 2를 출력
if a != b:
    print(2)
else:
    # a와 b가 같을 경우, a의 가장 작은 약수를 찾음
    i = 2
    while i * i <= a:  # i * i가 a보다 작거나 같을 때까지 반복
        if a % i == 0:  # a가 i로 나누어 떨어지면
            print(i)    # i가 가장 작은 약수이므로 출력하고 종료
            break
        i += 1
    else:
        # 만약 루프가 종료될 때까지 나누어 떨어지지 않으면 a는 소수
        print(a)  # a를 출력
